spec/mailers/application_mailer_spec.rb

```diff
@@ -0,0 +1,197 @@
+        check_sender { Person::AddRequestMailer.ask_person_to_add(request) }
+      end
+    end
+
+    describe Person::InactivityBlockMailer do
+      describe "#inactivity_block_warning" do
+        let(:recipient) { people(:bottom_member) }
+
+        it "has the sender per locale defined in the translation" do
+          check_sender { described_class.inactivity_block_warning(recipient) }
+        end
+      end
+    end
+
+    describe Person::LoginMailer do
+      let(:sender) { people(:top_leader) }
+      let(:recipient) { people(:bottom_member) }
+
+      it "has the sender per locale defined in the translation" do
+        check_sender { Person::LoginMailer.login(recipient, sender, "abcdef") }
+      end
+    end
+
+    describe Person::UserPasswordOverrideMailer do
+      let(:sender) { people(:top_leader) }
+      let(:recipient) { people(:bottom_member) }
+
+      it "has the sender per locale defined in the translation" do
+        check_sender { Person::UserPasswordOverrideMailer.send_mail(recipient, sender.full_name) }
+      end
+    end
+
+    describe DeliveryReportMailer do
+      let(:recipient_email) { "dude@hito42test.com" }
+      let(:envelope_sender) { "liste@hitobito.example.com" }
+      let(:mail_subject) { "Ausflugtips" }
+      let(:total_recipients) { 42 }
+      let(:delivered_at) { DateTime.now }
+      let(:formatted_delivered_at) { I18n.l(delivered_at) }
+
+      context "bulk mail success" do
+        let(:failed_recipients) { nil }
+
+        it "has the sender per locale defined in the translation" do
+          check_sender { DeliveryReportMailer.bulk_mail(recipient_email, envelope_sender, mail_subject, total_recipients, delivered_at) }
+        end
+      end
+    end
+
+    describe InvoiceMailer do
+      let(:invoice) { invoices(:invoice) }
+      let(:sender) { people(:bottom_member) }
+
+      it "has the sender per locale defined in the translation" do
+        check_sender { InvoiceMailer.notification(invoice, sender) }
+      end
+    end
+  end
+
+  describe InvoiceMailer do
+    let(:invoice) { invoices(:invoice) }
+    let(:sender) { people(:bottom_member) }
+
+    context "without translated sender" do
+      it "has the sender defined in the settings if the locale changes" do
+        check_sender(->(mail) { expect(mail[:from].value).to eq(Settings.email.sender) }) do
+          InvoiceMailer.notification(invoice, sender)
+        end
+      end
+    end
+
+    context "With broken sender translation" do
+      around do |example|
+        with_translations(
+          de: {settings: {email: {sender: "de broken email"}}},
+          fr: {settings: {email: {sender: "fr broken email"}}}
+        ) do
+          example.call
+        end
+      end
+
+      it "has the sender from the settings if the locale changes" do
+        check_sender(->(message) { expect(message[:from].value).to eq(Settings.email.sender) }) do
+          InvoiceMailer.notification(invoice, sender)
```

Comment:
I'm not sure how mail[:from].value here is not "de broken email". It's good, but I can't figure out how this happens. Mail:Message.from= does not do that.